central/graphql/resolvers/cluster_vulnerabilities_test.go

```diff
@@ -0,0 +1,603 @@
+	err = clusterCVEDatastore.UpsertClusterCVEsInternal(s.ctx, clusterCVEParts[0].CVE.Type, clusterCVEParts...)
+	s.NoError(err)
+}
+
+func (s *GraphQLClusterVulnerabilityTestSuite) TearDownSuite() {
+	s.envIsolator.RestoreAll()
+
+	clusterCVEPostgres.Destroy(s.ctx, s.db)
+	clusterCVEEdgePostgres.Destroy(s.ctx, s.db)
+	clusterPostgres.Destroy(s.ctx, s.db)
+	namespacePostgres.Destroy(s.ctx, s.db)
+	pgtest.CloseGormDB(s.T(), s.gormDB)
+	s.db.Close()
+}
+
+func (s *GraphQLClusterVulnerabilityTestSuite) TestUnauthorizedClusterVulnerabilityEndpoint() {
+	_, err := s.resolver.ClusterVulnerability(s.ctx, IDQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLClusterVulnerabilityTestSuite) TestUnauthorizedClusterVulnerabilitiesEndpoint() {
+	_, err := s.resolver.ClusterVulnerabilities(s.ctx, PaginatedQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLClusterVulnerabilityTestSuite) TestUnauthorizedClusterVulnerabilityCountEndpoint() {
+	_, err := s.resolver.ClusterVulnerabilityCount(s.ctx, RawQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLClusterVulnerabilityTestSuite) TestUnauthorizedClusterVulnerabilityCounterEndpoint() {
+	_, err := s.resolver.ClusterVulnerabilityCounter(s.ctx, RawQuery{})
+	s.Error(err, "Unauthorized request got through")
+}
+
+func (s *GraphQLClusterVulnerabilityTestSuite) TestClusterVulnerabilities() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	expectedIDs := []string{"clusterCve1", "clusterCve2", "clusterCve3", "clusterCve4", "clusterCve5"}
+	expectedCount := int32(len(expectedIDs))
+
+	vulns, err := s.resolver.ClusterVulnerabilities(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(expectedCount, int32(len(vulns)))
+	idList := getIDList(ctx, vulns)
+	s.ElementsMatch(expectedIDs, idList)
+
+	count, err := s.resolver.ClusterVulnerabilityCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(expectedCount, count)
+
+	counter, err := s.resolver.ClusterVulnerabilityCounter(ctx, RawQuery{})
+	s.NoError(err)
+	checkVulnerabilityCounter(s.T(), counter, expectedCount, 3, 1, 2, 1, 1)
+}
+
+func (s *GraphQLClusterVulnerabilityTestSuite) TestClusterVulnerabilitiesScoped() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	expectedIDs := []string{"clusterCve1", "clusterCve2", "clusterCve4", "clusterCve5"}
+	expectedCount := int32(len(expectedIDs))
+
+	cluster := s.getClusterResolver(ctx, s.clusterIDs[0])
+
+	vulns, err := cluster.ClusterVulnerabilities(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(expectedCount, int32(len(vulns)))
+	idList := getIDList(ctx, vulns)
+	s.ElementsMatch(expectedIDs, idList)
+
+	count, err := cluster.ClusterVulnerabilityCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(expectedCount, count)
+
+	counter, err := cluster.ClusterVulnerabilityCounter(ctx, RawQuery{})
+	s.NoError(err)
+	checkVulnerabilityCounter(s.T(), counter, expectedCount, 1, 1, 2, 0, 1)
```

Comment:
Fine for now but you definitely can refactor this out into a func and use it in required places.